src/es-module-shims.js

```diff
@@ -229,17 +209,39 @@ async function processScripts () {
     }
     else {
       importMapPromise = importMapPromise.then(async () =>
-        importMap = resolveAndComposeImportMap(script.src ? await (await fetch(script.src)).json() : JSON.parse(script.innerHTML), script.src || pageBaseUrl, importMap)
+        importMap = importShim.resolveImportMap(script.src ? await (await importShim.fetch(script.src)).json() : JSON.parse(script.innerHTML), script.src || pageBaseUrl, importMap)
       );
     }
     script.ep = true;
   }
 }
 
-function resolve (id, parentUrl) {
+function resolve (id, parentUrl, importMap) {
   return resolveImportMap(importMap, resolveIfNotPlainOrUrl(id, parentUrl) || id, parentUrl) || throwUnresolved(id, parentUrl);
 }
 
 function throwUnresolved (id, parentUrl) {
   throw Error("Unable to resolve specifier '" + id + (parentUrl ? "' from " + parentUrl : "'"));
 }
+
+const importShimOptions = self.importShimOptions || {};
+const importShimDefaults = {
+  fetch: url => fetch(url),
+  skip: /^https?:\/\/(cdn\.pika\.dev|dev\.jspm\.io|jspm\.dev)\//,
+  load: processScripts,
+  resolveImport: resolve,
+  resolveImportMap: resolveAndComposeImportMap,
```

Comment:
Rather than making this hook available, it could be nice to just handle this by hooking `load` entirely, and re-exposing the `importMap` property here for custom changes.

Reply:
the `load` hook IMO doesn't have much value, as there is too much closure based logic going on to make it reliably possible for a consumer to override. the intent of `resolveImportMap` is to simply hook into the logic of translating the json of a importmap-shim into a digestible format.  this will allow the json to be in a non-standard/compressed format but via the `resolveImportMap` be able to decompress into the format required.

Reply:
Are you saying the problem here is that you still want the `module-shim` scripts to load, while overriding the `importmap-shim` scripts?

Perhaps we could split load into two functions - `loadMaps` and `loadScripts`. Where those two functions are the top-level hooks that run one after the other.

Reply:
I'm looking for a hook that takes the data from `<script type="importmap-shim">` and then returns a `importMap` that `importShim` is familiar with (imports/scopes/depcache). This would allow the format of the data in `<script type="importmap-shim">` to be in *any* format, so long as the `importShim.resolveImportMap` mutates it to the expected shape.

```html
<script>
  importShim.resolveImportMap = (json, baseUrl, parentMap) => {
    // transform json to a new importMap, which would be the same shape as parentMap
    return newImportMap;
  };
</script>
<script type="importmap-shim" src="http://mycustom.importmap.format.json"></script>

Reply:
We must just have a single `loadMaps` function, it's only one line to do your own "search and fetch" for import maps.